### PR TITLE
Snowflake exam

### PR DESCRIPTION
**Pull request by zmaleki**
Creating a reference material for Snowpro core exam

### CODE REVIEW

File: _posts/2023-01-25-snowflake-snowpro-core-cert.md [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+SYSTEM | BLOCK sampling is often faster than BERNOULLI | ROW sampling. Also, BERNOULLI | ROW method is good for Smaller Tables, and SYSTEM | BLOCK method is for Larger Tables.
+
+### Snowflake architecture
+
+Snowflake's unique architecture consists of three key layers:
+ • Database Storage
+ • Query Processing
+ • Cloud Services
+
+## Some of the questions in my actual exam from back of my mind
```
Comment (temyers) [inline _posts/2023-01-25-snowflake-snowpro-core-cert.md:207]:
```suggestion
## Some sample questions
```

File: _posts/2023-01-25-snowflake-snowpro-core-cert.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+
+These are the questions that were not similar to udemy practice tests, hence I remember them
```
Comment (temyers) [inline _posts/2023-01-25-snowflake-snowpro-core-cert.md:209]: This sentence may need rewording slightly - 'here are some example questions you might face' or similar

Comment (temyers) [reply _posts/2023-01-25-snowflake-snowpro-core-cert.md:209]: "you can always refer to "these practise questions stick in my mind'"?

File: _posts/2023-01-25-snowflake-snowpro-core-cert.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+Relative_path and Modified_date
+- **How do you reference a sql variable?**
+using a $ sign
+- **How do you get the current version of the driver in SQL command?**
+SELECT CURRENT_CLIENT() SQL function
+- **What SQL ANSI constrains does Snowflake Enforce?**
+Snowflake supports defining and maintaining constraints like Primary, Foreign Key, Unique and NOT NULL, BUT does not enforce them except for NOT NULL constraints, which are always enforced.
+- **How many days does Snowflake keeps transiant tables in fail safe?**
+zero
+- **What's the retintion period of transient tables in time travel?**
```
Comment (bryan-yu) [inline _posts/2023-01-25-snowflake-snowpro-core-cert.md:221]: Spelt retention wrong

File: _posts/2023-01-25-snowflake-snowpro-core-cert.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+SYSTEM | BLOCK sampling is often faster than BERNOULLI | ROW sampling. Also, BERNOULLI | ROW method is good for Smaller Tables, and SYSTEM | BLOCK method is for Larger Tables.
+
+### Snowflake architecture
+
+Snowflake's unique architecture consists of three key layers:
+ • Database Storage
+ • Query Processing
+ • Cloud Services
+
+## Some sample questions
```
Comment (bryan-yu) [inline _posts/2023-01-25-snowflake-snowpro-core-cert.md:207]: super helpful 😄

**Review by RonakReyhani — COMMENTED** [review]
I love it Z, However my exam has a lot of questions around staging tables, what kind of staging table exist, the data cache in tables, who has access to which staging, i.e user stage table is only available to the user, but table stage to whom has access to the table. and also the schema tables and what each contains, off the top of my head the account_usage schema. 
I know we can't cover everything but these two was common ones I got in both practice and actual exam.

File: _posts/2023-01-25-snowflake-snowpro-core-cert.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+- **What are javascript stored procedure delimiters?**
+single quotes ' or double dollar signs \$\$.
+- **What are the output fields of directory table result?**
+Relative_path and Modified_date
```
Comment (bryan-yu) [inline _posts/2023-01-25-snowflake-snowpro-core-cert.md:212]: the full list currently is relative_path, size, last_modified, md5, file_url, etag. But guessing your exam q had only those 2 main ones. (I didnt get this question )

File: _posts/2023-01-25-snowflake-snowpro-core-cert.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+**UDF FUNCTIONS** -> With user-defined functions (UDFs), you can extend the system to perform operations that are not available through the built-in, system-defined functions provided by Snowflake.
+
+**AGGREGATE FUNCTIONS** -> An aggregate function takes multiple rows (actually, zero, one, or more rows) as input and produces a single output. In contrast, scalar functions take one row as input and produce one row (one value) as output.
+
+### Snowflake built-in views and tables
+
+**AUTOMATIC_CLUSTERING_HISTORY** table -> Snowflake Information Schema:
+view the billing for Automatic Clustering
+
+**AUTOMATIC_CLUSTERING_HISTORY** View -> Account Usage:
```
Comment (bryan-yu) [inline _posts/2023-01-25-snowflake-snowpro-core-cert.md:111]: minor, mm bit of inconsistency with the View and view uppercase/lowercase spelling in this section as well as for REPLICATION_USAGE_HISTORY and REPLICATION_USAGE_HISTORY you have it as `Account Usage schema` and `Snowflake Information Schema`. But for the rest you got `INFORMATION_SCHEMA` and `Account Usage`

File: _posts/2023-01-25-snowflake-snowpro-core-cert.md [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+date: 2023-01-25
+highlight: monokai
+image: /img/skeleton-loaders/banner.jpg
+author: Zainab Maleki
+tags: [snowflake, snowpro, core, certification, cheatsheet, exam, notes]
+---
+
+# Snowflake SnowPro Core Certification cheat sheet
+
+After working with Snowflake for almost four years I have finally bowed down to studying for the Snowflake certification and passed my exam this morning. Before I started going through the practice tests I was thinking with four years of experience passing the exam should be fairly easy. However soon after going through my first online practice test, I realized the practical experience of using Snowflake plus intuition and some luck would probably only get you less than 50% of the score and you will need a lot of studying for things that you don't normally use or you would google whenever you come across those requirements. Unfortunately googling is not an option during the test, hence this cheat sheet might come handy if you are studying for this cert.
```
Comment (bryan-yu) [inline _posts/2023-01-25-snowflake-snowpro-core-cert.md:18]: - `with four years of experience passing, the exam should be fairly easy`. Add a comma between `passing` and `the exam`.
-

**Comment by zmaleki** [comment]
> I love it Z, However my exam has a lot of questions around staging tables, what kind of staging table exist, the data cache in tables, who has access to which staging, i.e user stage table is only available to the user, but table stage to whom has access to the table. and also the schema tables and what each contains, from the top of my head the account_usage schema. I know we can't cover everything but these two was common ones I got in both practice and actual exam.

Oh thanks Nadia. Yea I did add a reference for directory tables which is basically where information is kept about staged files. I also added my notes on the stage file functions which was the most confusing one for me. But I tried to only focus on questions and notes that I consistently kept getting wrong, otherwise it would become a really long blog post.

**Review by temyers — APPROVED** [review]
LGTM!

**Review by bryan-yu — APPROVED** [review]

**Pull request merged**